lib/puppet/provider/package/dnf.rb

```diff
@@ -28,7 +28,7 @@
       end
   end
 
-  defaultfor :operatingsystem => :fedora, :operatingsystemmajrelease => ['22', '23', '24', '25']
+  defaultfor :operatingsystem => :fedora, :operatingsystemmajrelease => (22..30).to_a
```

Comment:
Clearly this is the simplest thing to do, but I'm wondering if it might be better here to use a feature (which allows us to execute an arbitrary check) rather than to hardcode version numbers.

I haven't tested, but it would look something like this:

```ruby
defaultfor :operatingsystem => :fedora, :feature => supports_dnf
```


There's an example of this in the posix exec provider at https://github.com/puppetlabs/puppet/blob/master/lib/puppet/provider/exec/posix.rb#L6 which relies on the `:posix` feature definition at https://github.com/puppetlabs/puppet/blob/937505096316f2389d5ba0c7f13b3da4eef36e0f/lib/puppet/feature/base.rb#L11

The `:feature` support was fixed as part of PUP-1766 / https://github.com/puppetlabs/puppet/commit/784b5e6090ea9090c630055fcd940e8a8f2d4f05 and lives at https://github.com/puppetlabs/puppet/blob/bf3d94cc50a941d5e4d413568dbf611e95848eb9/lib/puppet/provider.rb#L285